### ISSUE TITLE
UWP and Android NavBar problem for MasterDetail Page

### ISSUE THREAD

**Issue by wincotv** [xamarin/Xamarin.Forms#4292]
### Description

I have defined NavBar in MasterDetailPage
![image](https://user-images.githubusercontent.com/3726570/47950649-481eea80-df55-11e8-894a-e968aa59c393.png)

With UWP and version XF 3.3.0.912540 didn't work right align but NavBar was visible 
![image](https://user-images.githubusercontent.com/3726570/47950692-0b072800-df56-11e8-85b8-db02b1f96d9c.png)

After update to 3.3.0.967583 I can't see NavBar 
![image](https://user-images.githubusercontent.com/3726570/47950740-ca5bde80-df56-11e8-8c8c-b4e3f1c8d99b.png)

On Android it same with both version XF, I can't see NavBar when it defined in MasterDetailPage.

**Comment by PureWeen** [comment]
@wincotv  if you define the titleview inside the currently navigated to page does it show up how you'd expect it to?

**Comment by wincotv** [comment]
@PureWeen , yes now work also for UWP if I have defined TitleView in Page or in code , with ok right align, but resize form on UWP don't refresh navbar.
![image](https://user-images.githubusercontent.com/3726570/47962414-b8dd0a00-e01c-11e8-936d-4ff1fa864af8.png)
![image](https://user-images.githubusercontent.com/3726570/47962433-d27e5180-e01c-11e8-846f-c8d1a74bb16c.png)

**Comment by PureWeen** [comment]
@wincotv  The scenario here is a bit unfortunate

When titleview was first implemented you could specify the titleview on the navigation page itself but that only worked on UWP. Then when I took it over and did my set of changes I removed that ability on UWP without realizing it because the other platforms didn't let you do that so I didn't notice anything had changed.  I realize I regressed UWP but right now it's consistent between the three platforms which is more ideal.

I'm going to move this issue to an enhancement for TitleView to allow specifying it on the NavigationPage itself so that it works on all three platforms

**Comment by stevenfowler16** [comment]
First time in Xamarin today and I just spent an hour trying to figure out why my titleview wasn't working.   Is this an easy issue to fix?
